confgenerator/confgenerator.go

```
@@ -37,262 +36,70 @@ func defaultFilepathJoin(_ string, elem ...string) string {
-	receiverNameMap := make(map[string]string)
-	hostmetricsReceiverFactories, mssqlReceiverFactories, iisReceiverFactories, err := extractOtelReceiverFactories(receivers)
-	if err != nil {
-		return nil, nil, nil, nil, err
-	}
-	for _, pID := range sortedKeys(pipelines) {
-		p := pipelines[pID]
+func (m *Metrics) generateOtelPipelines() (map[string]otel.Pipeline, error) {
+	out := make(map[string]otel.Pipeline)
+	for pID, p := range m.Service.Pipelines {
 		for _, rID := range p.ReceiverIDs {
-			if _, ok := receiverNameMap[rID]; ok {
-				continue
-			}
-			if h, ok := hostmetricsReceiverFactories[rID]; ok {
-				hostMetrics := otel.HostMetrics{
-					HostMetricsID:      rID,
-					CollectionInterval: h.CollectionInterval,
-				}
-				hostMetricsList = append(hostMetricsList, &hostMetrics)
-				receiverNameMap[rID] = "hostmetrics/" + rID
-			} else if m, ok := mssqlReceiverFactories[rID]; ok {
-				mssql := otel.MSSQL{
-					MSSQLID:            rID,
-					CollectionInterval: m.CollectionInterval,
-				}
-				mssqlList = append(mssqlList, &mssql)
-				receiverNameMap[rID] = "windowsperfcounters/mssql_" + rID
-			} else if i, ok := iisReceiverFactories[rID]; ok {
-				iis := otel.IIS{
-					IISID:              rID,
-					CollectionInterval: i.CollectionInterval,
-				}
-				iisList = append(iisList, &iis)
-				receiverNameMap[rID] = "windowsperfcounters/iis_" + rID
-			}
-		}
-	}
-	return hostMetricsList, mssqlList, iisList, receiverNameMap, nil
-}
-
-func generateOtelExporters(exporters map[string]MetricsExporter, pipelines map[string]*MetricsPipeline) ([]*otel.Stackdriver, map[string]string, error) {
-	stackdriverList := []*otel.Stackdriver{}
-	exportNameMap := make(map[string]string)
-	for _, pID := range sortedKeys(pipelines) {
-		p := pipelines[pID]
-		for _, eID := range p.ExporterIDs {
-			exporter, ok := exporters[eID]
+			receiver, ok := m.Receivers[rID]
 			if !ok {
-				continue
+				return nil, fmt.Errorf("receiver %q not found", rID)
```

It would be nice if we could report this error using that line number + column format.